h.265 QSV does not respond to changes of ICQ

With QSV h.265 10 bit it seems like Handbrake 1.6 does not respond to changes in ICQ. Filesize and bitrate remain unchanged no matter if I chose ICQ 21 or 2. On the the other hand both parameters change if I vary average bitrate.

Thank you for the hint with the old driver. I updated accordingly and repeated the tests with the new driver. Unfortunatly there is no change: Filesizes remain unchanged no matter which ICQ value I chose. I attach again two logfiles with ICQ2 and 20. In both cases the mkv file size comes out with each 6.329 kb. Again changes in average bitrate lead to different file size (see third attached file).

 

It seems the QSV 10bit h.265 encoder is locked to a bitrate of 790 kb/s no matter which value of ICQ is chosen.

 

Thanks again for looking into this

 

Von: Scott ***@***.***> 
Gesendet: Samstag, 7. Januar 2023 14:17
An: HandBrake/HandBrake ***@***.***>
Cc: dita61 ***@***.***>; Author ***@***.***>
Betreff: Re: [HandBrake/HandBrake] h.265 QSV does not respond to changes of ICQ (Issue #4814)

 

Your GPU driver is not up-to-date. It's an old unsupported version.

https://www.intel.co.uk/content/www/uk/en/products/sku/196597/intel-core-i71065g7-processor-8m-cache-up-to-3-90-ghz/downloads.html

Unfortunately, if the vendor of your machine has vendor locked their version and isn't updating it, then your pretty much out of luck. All I can suggest is maybe try older versions of HandBrake.

—
Reply to this email directly, view it on GitHub <https://github.com/HandBrake/HandBrake/issues/4814#issuecomment-1374478574> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/A5DUXX22IP26MSUWM4BBXLLWRFUFZANCNFSM6AAAAAATTIMAV4> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/A5DUXX6IO2MFNAU4Q65TV6LWRFUFZA5CNFSM6AAAAAATTIMAV6WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSR5TQO4.gif> Message ID: ***@***.*** ***@***.***> >


HandBrake 1.6.0 (2022122800)
OS: Microsoft Windows NT 10.0.19045.0
CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
Ram: 16153 MB, 
GPU Information:
  Intel(R) Iris(R) Plus Graphics - 31.0.101.2115
Screen: 1280x800
Temp Dir: C:\Users\Dieter\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\Dieter\AppData\Roaming\HandBrake

-------------------------------------------
[13:27:50] hb_init: starting libhb thread

 # Starting Scan ...

[13:27:50] CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
[13:27:50]  - Intel microarchitecture Ice Lake
[13:27:50]  - logical processor count: 8
[13:27:50] Intel Quick Sync Video support: yes, gpu list: 0
[13:27:50] Intel Quick Sync Video integrated adapter with index 0
[13:27:50] Impl mfxhw64 library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_b53c057d22ce6f37\libmfxhw64.dll
[13:27:50]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[13:27:50]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes)
[13:27:50]  - H.264 encoder: yes
[13:27:50]     - preferred implementation: hardware (1) via D3D11
[13:27:50]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+repeatpps+ib_adapt+nmpslice
[13:27:50]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[13:27:50]     - preferred implementation: hardware (1) via D3D11
[13:27:50]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1
[13:27:50]  - AV1 encoder: no
[13:27:50] hb_scan: path=D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts, title_index=0
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
[13:27:50] bd: not a bd - trying as a stream/file instead
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[13:27:50] dvd: not a dvd - trying as a stream/file instead
[13:27:50] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:27:50] Found the following PIDS
[13:27:50]     Video PIDS :
[13:27:50]       0x17de type H.264 (0x1b) (PCR)
[13:27:50]     Audio PIDS :
[13:27:50]       0x17e8 type AC3 (0x81)
[13:27:50]     Subtitle PIDS :
[13:27:50]     Other PIDS :
[13:27:50] stream id 0x17e8 (type 0x81 substream 0x0) audio 0x17e8
[13:27:50] scan: decoding previews for title 1
[13:27:50] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:27:50] scan: audio 0x17e8: ac3, rate=48000Hz, bitrate=448000 Deutsch (AC3) (5.1 ch) (448 kbps)
[13:27:51] stream: 29 good frames, 0 errors (0%)
[13:27:51] scan: 10 previews, 1280x720, 50.000 fps, autocrop = 0/2/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1, chroma location: left
[13:27:51] scan: supported video decoders: avcodec qsv
[13:27:51] libhb: scan thread found 1 valid title(s)

 # Scan Finished ...


HandBrake 1.6.0 (2022122800)
OS: Microsoft Windows NT 10.0.19045.0
CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
Ram: 16153 MB, 
GPU Information:
  Intel(R) Iris(R) Plus Graphics - 31.0.101.2115
Screen: 1280x800
Temp Dir: C:\Users\Dieter\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\Dieter\AppData\Roaming\HandBrake

-------------------------------------------

 # Starting Encode ...

[13:35:48] base preset: h265 Intel QSV (Modified)
[13:35:48] Remote Process started with Process ID: 10656 using port: 8037. Max Allowed Instances: 1
[13:35:48] Worker: Starting HandBrake Engine ...
[13:35:48] Worker: Starting Web Server on port 8037 ...
[13:35:49] Worker: Disconnected worker monitoring enabled!
[13:35:49] Compile-time hardening features are enabled
[13:35:49] qsv: is available on this system
[13:35:49] vcn: not available on this system
Cannot load nvEncodeAPI64.dll
[13:35:49] hb_init: starting libhb thread
[13:35:49] Starting work at: Sun Jan 08 13:35:49 2023
[13:35:49] 1 job(s) to process
[13:35:49] json job:
{
  "Audio": {
    "AudioList": [],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:mp3",
      "copy:truehd",
      "copy:flac",
      "copy:eac3"
    ]
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "D:\\Filme schnitt\\test billboard\\TestHB1.6_ICQ20.mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 14,
        "Settings": {
          "crop-bottom": "2",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "0",
          "height": "718",
          "width": "1280"
        }
      },
      {
        "ID": 7,
        "Settings": {
          "mode": "0"
        }
      }
    ]
  },
  "PAR": {
    "Num": 1,
    "Den": 1
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "D:\\Filme schnitt\\test billboard\\Test Three Billboards Outside Ebbing, Missoury_fixed.ts"
  },
  "Subtitle": {
    "Search": {
      "Burn": true,
      "Default": false,
      "Enable": true,
      "Forced": true
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h265_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "lowpower=1",
    "Preset": "quality",
    "Profile": "auto",
    "Quality": 20,
    "QSV": {
      "Decode": true
    },
    "HardwareDecode": 0
  }
}
[13:35:49] CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
[13:35:49]  - Intel microarchitecture Ice Lake
[13:35:49]  - logical processor count: 8
[13:35:49] Intel Quick Sync Video support: yes, gpu list: 0
[13:35:49] Intel Quick Sync Video integrated adapter with index 0
[13:35:49] Impl mfxhw64 library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_b53c057d22ce6f37\libmfxhw64.dll
[13:35:49]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[13:35:49]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes)
[13:35:49]  - H.264 encoder: yes
[13:35:49]     - preferred implementation: hardware (1) via D3D11
[13:35:49]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+repeatpps+ib_adapt+nmpslice
[13:35:49]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[13:35:49]     - preferred implementation: hardware (1) via D3D11
[13:35:49]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1
[13:35:49]  - AV1 encoder: no
[13:35:49] hb_scan: path=D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
[13:35:49] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdlibdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
vdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[13:35:49] dvd: not a dvd - trying as a stream/file instead
[13:35:49] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:35:49] Found the following PIDS
[13:35:49]     Video PIDS :
[13:35:49]       0x17de type H.264 (0x1b) (PCR)
[13:35:49]     Audio PIDS :
[13:35:49]       0x17e8 type AC3 (0x81)
[13:35:49]     Subtitle PIDS :
[13:35:49]     Other PIDS :
[13:35:49] stream id 0x17e8 (type 0x81 substream 0x0) audio 0x17e8
[13:35:49] scan: decoding previews for title 1
[13:35:49] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:35:49] scan: audio 0x17e8: ac3, rate=48000Hz, bitrate=448000 Deutsch (AC3) (5.1 ch) (448 kbps)
[13:35:50] stream: 29 good frames, 0 errors (0%)
[13:35:50] scan: 10 previews, 1280x720, 50.000 fps, autocrop = 0/2/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1, chroma location: left
[13:35:50] scan: supported video decoders: avcodec qsv
[13:35:50] libhb: scan thread found 1 valid title(s)
[13:35:50] qsv: integrated qsv adapter with index 0 has been selected
[13:35:50] Skipping subtitle scan.  No suitable subtitle tracks.
[13:35:50] Starting Task: Encoding Pass
[13:35:50] Skipping vfr filter
[13:35:50] work: only 1 chapter, disabling chapter markers
[13:35:50] job configuration:
[13:35:50]  * source
[13:35:50]    + D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts
[13:35:50]    + title 1, chapter(s) 1 to 1
[13:35:50]  * destination
[13:35:50]    + D:\Filme schnitt\test billboard\TestHB1.6_ICQ20.mkv
[13:35:50]    + container: Matroska (libavformat)
[13:35:50]  * video track
[13:35:50]    + decoder: h264_qsv 8-bit (yuv420p)
[13:35:50]      + bitrate 200 kbps
[13:35:50]    + filters
[13:35:50]      + Crop and Scale (width=1280:height=718:crop-top=0:crop-bottom=2:crop-left=0:crop-right=0)
[13:35:50]        + source: 1280 * 720, crop (0/2/0/0): 1280 * 718, scale: 1280 * 718
[13:35:50]      + Format (format=p010le)
[13:35:50]    + Output geometry
[13:35:50]      + storage dimensions: 1280 x 718
[13:35:50]      + pixel aspect ratio: 1 : 1
[13:35:50]      + display dimensions: 1280 x 718
[13:35:50]    + encoder: H.265 10-bit (Intel Media SDK)
[13:35:50]      + preset:  quality
[13:35:50]      + options: lowpower=1
[13:35:50]      + profile: auto
[13:35:50]      + level:   auto
[13:35:50]      + quality: 20.00 (ICQ)
[13:35:50]      + color profile: 1-1-1
[13:35:50]      + chroma location: left
[13:35:50] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:35:50] sync: expecting 3184 video frames
[13:35:50] encqsvInit: using encode-only (LowPower) path
[13:35:50] encqsvInit: H.265/HEVC Main 10 profile @ level (null)
[13:35:50] encqsvInit: TargetUsage 1 AsyncDepth 6
[13:35:50] encqsvInit: GopRefDist 1 GopPicSize 52 NumRefFrame 3 IdrInterval 0
[13:35:50] encqsvInit: BFramesMax 0 BRefType pyramid
[13:35:50] encqsvInit: RateControlMethod ICQ ICQQuality 20
[h264_qsv @ 000001640ab40b00] Invalid pkt_timebase, passing timestamps as-is.
[13:35:51] sync: first pts video is 0
[13:35:51] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.35
[13:36:02] sync: video time went backwards 100 ms, dropped 5 frames. PTS 5270554
[13:36:03] hb_ts_stream_decode - eof
[13:36:03] hb_ts_stream_decode - eof
[13:36:03] reader: done. 2 scr changes
[13:36:03] work: average encoding speed for job is 258.813202 fps
[13:36:03] stream: 3212 good frames, 0 errors (0%)
[13:36:03] h264_qsv-decoder done: 3212 frames, 0 decoder errors
[13:36:03] sync: got 3207 frames, 3184 expected
[13:36:03] sync: framerate min 8.333 fps, max 50.000 fps, avg 49.888 fps
[13:36:03] mux: track 0, 3207 frames, 6454839 bytes, 803.34 kbps, fifo 4096
[13:36:03] Finished work at: Sun Jan 08 13:36:03 2023
[13:36:03] libhb: work result = 0

 # Job Completed!



HandBrake 1.6.0 (2022122800)
OS: Microsoft Windows NT 10.0.19045.0
CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
Ram: 16153 MB, 
GPU Information:
  Intel(R) Iris(R) Plus Graphics - 31.0.101.2115
Screen: 1280x800
Temp Dir: C:\Users\Dieter\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\Dieter\AppData\Roaming\HandBrake

-------------------------------------------

 # Starting Encode ...

[13:56:08] base preset: h265 Intel QSV (Modified)
[13:56:08] Remote Process started with Process ID: 12460 using port: 8039. Max Allowed Instances: 1
[13:56:08] Worker: Starting HandBrake Engine ...
[13:56:08] Worker: Starting Web Server on port 8039 ...
[13:56:09] Worker: Disconnected worker monitoring enabled!
[13:56:09] Compile-time hardening features are enabled
[13:56:09] qsv: is available on this system
[13:56:09] vcn: not available on this system
Cannot load nvEncodeAPI64.dll
[13:56:09] hb_init: starting libhb thread
[13:56:09] Starting work at: Sun Jan 08 13:56:09 2023
[13:56:09] 1 job(s) to process
[13:56:09] json job:
{
  "Audio": {
    "AudioList": [],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:mp3",
      "copy:truehd",
      "copy:flac",
      "copy:eac3"
    ]
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "D:\\Filme schnitt\\test billboard\\TestHB1.6_AVBR2800.mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 14,
        "Settings": {
          "crop-bottom": "2",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "0",
          "height": "718",
          "width": "1280"
        }
      },
      {
        "ID": 7,
        "Settings": {
          "mode": "0"
        }
      }
    ]
  },
  "PAR": {
    "Num": 1,
    "Den": 1
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "D:\\Filme schnitt\\test billboard\\Test Three Billboards Outside Ebbing, Missoury_fixed.ts"
  },
  "Subtitle": {
    "Search": {
      "Burn": true,
      "Default": false,
      "Enable": true,
      "Forced": true
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h265_10bit",
    "Level": "auto",
    "Bitrate": 2800,
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "lowpower=1",
    "Preset": "quality",
    "Profile": "auto",
    "QSV": {
      "Decode": true
    },
    "HardwareDecode": 0
  }
}
[13:56:09] CPU: Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
[13:56:09]  - Intel microarchitecture Ice Lake
[13:56:09]  - logical processor count: 8
[13:56:09] Intel Quick Sync Video support: yes, gpu list: 0
[13:56:09] Intel Quick Sync Video integrated adapter with index 0
[13:56:09] Impl mfxhw64 library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_b53c057d22ce6f37\libmfxhw64.dll
[13:56:09]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[13:56:09]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes)
[13:56:09]  - H.264 encoder: yes
[13:56:09]     - preferred implementation: hardware (1) via D3D11
[13:56:09]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+repeatpps+ib_adapt+nmpslice
[13:56:09]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[13:56:09]     - preferred implementation: hardware (1) via D3D11
[13:56:09]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1
[13:56:09]  - AV1 encoder: no
[13:56:09] hb_scan: path=D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
[13:56:09] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdvlibdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
dread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[13:56:09] dvd: not a dvd - trying as a stream/file instead
[13:56:09] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:56:09] Found the following PIDS
[13:56:09]     Video PIDS :
[13:56:09]       0x17de type H.264 (0x1b) (PCR)
[13:56:09]     Audio PIDS :
[13:56:09]       0x17e8 type AC3 (0x81)
[13:56:09]     Subtitle PIDS :
[13:56:09]     Other PIDS :
[13:56:09] stream id 0x17e8 (type 0x81 substream 0x0) audio 0x17e8
[13:56:09] scan: decoding previews for title 1
[13:56:09] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:56:10] scan: audio 0x17e8: ac3, rate=48000Hz, bitrate=448000 Deutsch (AC3) (5.1 ch) (448 kbps)
[13:56:10] stream: 29 good frames, 0 errors (0%)
[13:56:10] scan: 10 previews, 1280x720, 50.000 fps, autocrop = 0/2/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1, chroma location: left
[13:56:10] scan: supported video decoders: avcodec qsv
[13:56:10] libhb: scan thread found 1 valid title(s)
[13:56:10] qsv: integrated qsv adapter with index 0 has been selected
[13:56:10] Skipping subtitle scan.  No suitable subtitle tracks.
[13:56:10] Starting Task: Encoding Pass
[13:56:10] Skipping vfr filter
[13:56:10] work: only 1 chapter, disabling chapter markers
[13:56:10] job configuration:
[13:56:10]  * source
[13:56:10]    + D:\Filme schnitt\test billboard\Test Three Billboards Outside Ebbing, Missoury_fixed.ts
[13:56:10]    + title 1, chapter(s) 1 to 1
[13:56:10]  * destination
[13:56:10]    + D:\Filme schnitt\test billboard\TestHB1.6_AVBR2800.mkv
[13:56:10]    + container: Matroska (libavformat)
[13:56:10]  * video track
[13:56:10]    + decoder: h264_qsv 8-bit (yuv420p)
[13:56:10]      + bitrate 200 kbps
[13:56:10]    + filters
[13:56:10]      + Crop and Scale (width=1280:height=718:crop-top=0:crop-bottom=2:crop-left=0:crop-right=0)
[13:56:10]        + source: 1280 * 720, crop (0/2/0/0): 1280 * 718, scale: 1280 * 718
[13:56:10]      + Format (format=p010le)
[13:56:10]    + Output geometry
[13:56:10]      + storage dimensions: 1280 x 718
[13:56:10]      + pixel aspect ratio: 1 : 1
[13:56:10]      + display dimensions: 1280 x 718
[13:56:10]    + encoder: H.265 10-bit (Intel Media SDK)
[13:56:10]      + preset:  quality
[13:56:10]      + options: lowpower=1
[13:56:10]      + profile: auto
[13:56:10]      + level:   auto
[13:56:10]      + bitrate: 2800 kbps, pass: 0
[13:56:10]      + color profile: 1-1-1
[13:56:10]      + chroma location: left
[13:56:10] file is MPEG Transport Stream with 188 byte packets offset 0 bytes
[13:56:10] sync: expecting 3184 video frames
[13:56:10] encqsvInit: using encode-only (LowPower) path
[13:56:10] encqsvInit: H.265/HEVC Main 10 profile @ level 4.0
[13:56:10] encqsvInit: TargetUsage 1 AsyncDepth 6
[13:56:10] encqsvInit: GopRefDist 1 GopPicSize 52 NumRefFrame 3 IdrInterval 0
[13:56:10] encqsvInit: BFramesMax 0 BRefType pyramid
[13:56:10] encqsvInit: RateControlMethod VBR TargetKbps 2800 MaxKbps 2800 BufferSizeInKB 700 InitialDelayInKB 350 BRCParamMultiplier 1
[h264_qsv @ 0000023c5f80a740] Invalid pkt_timebase, passing timestamps as-is.
[13:56:11] sync: first pts video is 0
[13:56:11] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.35
[13:56:23] sync: video time went backwards 100 ms, dropped 5 frames. PTS 5270554
[13:56:24] hb_ts_stream_decode - eof
[13:56:24] hb_ts_stream_decode - eof
[13:56:24] reader: done. 2 scr changes
[13:56:24] work: average encoding speed for job is 234.661896 fps
[13:56:24] stream: 3212 good frames, 0 errors (0%)
[13:56:24] h264_qsv-decoder done: 3212 frames, 0 decoder errors
[13:56:25] sync: got 3207 frames, 3184 expected
[13:56:25] sync: framerate min 8.333 fps, max 50.000 fps, avg 49.888 fps
[13:56:25] mux: track 0, 3207 frames, 22165786 bytes, 2758.65 kbps, fifo 2048
[13:56:25] Finished work at: Sun Jan 08 13:56:25 2023
[13:56:25] libhb: work result = 0

 # Job Completed!

--- a/NEWS.markdown
+++ b/NEWS.markdown
@@ -63,11 +63,17 @@ Windows users, please make sure to install [Microsoft .NET Desktop Runtime versi
 - Various fixes and library updates for QuickSync to improve support on Linux (#4958)
 - Switch to using swscale instead of zscale when the resolution isn't mod2. Should fix scan failures in this condition
 
+### Linux
+
+- Fixed multiple QSV device enumeration issue and enable Intel QSV AV1 (hardware) video encoders using Flatpak Intel Media plugin (#4775)
+- Fixed default usage of QSV lowpower mode causing issues if HuC is disabled (#4768)
+
 ### Windows
 
 - Fixed an issue with autonaming when using drive based sources (#4859)
 - Fixed Title Specific Scan for drive sources. (#4921)
 - Fix an issue that could cause a preset to show as "modified" when it was not. (#4909, #4908)
+- Fixed an issue with QSV bitrate does not respond to changes of ICQ (#4814)
 - Some reliability improvements in the Process Isolation Feature. 
 - Miscellaneous other fixes
 

--- a/libhb/enc_qsv.c
+++ b/libhb/enc_qsv.c
@@ -1627,9 +1627,9 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
     // set the keyframe interval
     if (pv->param.gop.gop_pic_size < 0)
     {
-        int rate = (double)job->orig_vrate.num / job->orig_vrate.den + 0.5;
+        double rate = (double)job->orig_vrate.num / job->orig_vrate.den + 0.5;
         // set the keyframe interval based on the framerate
-        pv->param.gop.gop_pic_size = rate;
+        pv->param.gop.gop_pic_size = (int)(FFMIN(rate * 2, 120));
     }
     pv->param.videoParam->mfx.GopPicSize = pv->param.gop.gop_pic_size;
 

--- a/libhb/enc_qsv.c
+++ b/libhb/enc_qsv.c
@@ -1452,8 +1452,11 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
         }
     }
 
+    int hw_generation = hb_qsv_hardware_generation(hb_qsv_get_platform(hb_qsv_get_adapter_index()));
     // sanitize ICQ
-    if (!(pv->qsv_info->capabilities & HB_QSV_CAP_RATECONTROL_ICQ))
+    // workaround for MediaSDK platforms below TGL to disable ICQ if incorrectly detected
+    if (!(pv->qsv_info->capabilities & HB_QSV_CAP_RATECONTROL_ICQ) ||
+        ((pv->param.videoParam->mfx.LowPower == MFX_CODINGOPTION_ON) && (hw_generation < QSV_G8)))
     {
         // ICQ not supported
         pv->param.rc.icq = 0;
@@ -1608,7 +1611,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
     // set the GOP structure
     if (pv->param.gop.gop_ref_dist < 0)
     {
-        if ((hb_qsv_hardware_generation(hb_qsv_get_platform(hb_qsv_get_adapter_index())) >= QSV_G8) &&
+        if ((hw_generation >= QSV_G8) &&
             (pv->param.videoParam->mfx.CodecId == MFX_CODEC_HEVC ||
             pv->param.videoParam->mfx.CodecId == MFX_CODEC_AV1))
         {

--- a/libhb/handbrake/av1_common.h
+++ b/libhb/handbrake/av1_common.h
@@ -15,22 +15,10 @@
 static const char * const hb_av1_profile_names[] = {
     "auto", "main", "high", "professional", NULL, };
 
-#if HB_PROJECT_FEATURE_QSV
-static const char * const hb_av1_qsv_profile_names[] = {
-    "auto", "main", NULL, };
-#endif
-
 static const char * const hb_av1_level_names[] = {
     "auto", "2.0", "2.1", "2.2", "2.3", "3.0", "3.1", "3.2",
     "3.3", "4.0", "4.1", "4.2", "4.3", "5.0", "5.1", "5.2",
     "5.3", "6.0", "6.1", "6.2", "6.3", NULL, };
-
-#if HB_PROJECT_FEATURE_QSV
-static const char * const hb_av1_qsv_level_names[] = {
-    "auto", "2.0", "2.1", "2.2", "2.3", "3.0", "3.1", "3.2",
-    "3.3", "4.0", "4.1", "4.2", "4.3", "5.0", "5.1", "5.2",
-    "5.3", "6.0", NULL, };
-#endif
 
 static const int          hb_av1_level_values[] = {
      -1,  20,  21,  22,  23,  30,  31,  32,  33,  40,  41,  42,

--- a/libhb/handbrake/h265_common.h
+++ b/libhb/handbrake/h265_common.h
@@ -26,10 +26,6 @@ static const char * const hb_h265_profile_names_10bit[]     = {
     "auto", "main10", "main10-intra", NULL, };
 static const char * const hb_x265_profile_names_10bit[]     = {
     "auto", "main10", "main10-intra", "main422-10", "main422-10-intra", "main444-10", "main444-10-intra", NULL, };
-#if HB_PROJECT_FEATURE_QSV
-static const char * const hb_h265_qsv_profile_names_10bit[] = {
-    "auto", "main10", NULL, };
-#endif
 static const char * const hb_h265_profile_names_12bit[]     = {
     "auto", "main12", "main12-intra", NULL, };
 static const char * const hb_x265_profile_names_12bit[]     = {

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -3602,10 +3602,12 @@ int hb_qsv_param_default(hb_qsv_param_t *param, mfxVideoParam *videoParam,
         {
             param->videoParam->ExtParam[param->videoParam->NumExtParam++] = (mfxExtBuffer*)&param->av1BitstreamParam;
         }
+#if defined(_WIN32) || defined(__MINGW32__)
         if (info->capabilities & HB_QSV_CAP_LOWPOWER_ENCODE)
         {
             param->videoParam->mfx.LowPower = MFX_CODINGOPTION_ON;
         }
+#endif
     }
     else
     {

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -124,12 +124,23 @@ static hb_triplet_t hb_qsv_h265_profiles[] =
     { "Main Still Picture", "mainstillpicture", MFX_PROFILE_HEVC_MAINSP, },
     { NULL,                                                              },
 };
+static const char * const hb_qsv_h265_profiles_names_10bit[] = {
+    "auto",
+    "main10",
+    NULL,
+};
 static hb_triplet_t hb_qsv_av1_profiles[] =
 {
     { "Main",               "main",             MFX_PROFILE_AV1_MAIN,    },
     { "High",               "high",             MFX_PROFILE_AV1_HIGH,    },
     { "Professional",       "professional",     MFX_PROFILE_AV1_PRO,     },
     { NULL,                                                              },
+};
+static const char * const hb_qsv_av1_profiles_names[] =
+{
+    "auto",
+    "main",
+    NULL,
 };
 static hb_triplet_t hb_qsv_vpp_scale_modes[] =
 {
@@ -172,6 +183,11 @@ static hb_triplet_t hb_qsv_h264_levels[] =
     { "5.2", "5.2", MFX_LEVEL_AVC_52, },
     { NULL,                           },
 };
+static const char * const hb_h264_qsv_level_names[] =
+{
+    "auto", "1.0", "1b", "1.1", "1.2", "1.3", "2.0", "2.1", "2.2", "3.0",
+    "3.1", "3.2", "4.0", "4.1", "4.2", "5.0", "5.1", "5.2", NULL,
+};
 static hb_triplet_t hb_qsv_h265_levels[] =
 {
     { "1.0", "1.0", MFX_LEVEL_HEVC_1,  },
@@ -188,6 +204,11 @@ static hb_triplet_t hb_qsv_h265_levels[] =
     { "6.1", "6.1", MFX_LEVEL_HEVC_61, },
     { "6.2", "6.2", MFX_LEVEL_HEVC_62, },
     { NULL,                            },
+};
+static const char * const hb_h265_qsv_level_names[] =
+{
+    "auto", "1.0", "2.0", "2.1", "3.0", "3.1", "4.0", "4.1",
+    "5.0", "5.1", "5.2", "6.0", "6.1", "6.2", NULL,
 };
 static hb_triplet_t hb_qsv_av1_levels[] =
 {
@@ -207,11 +228,13 @@ static hb_triplet_t hb_qsv_av1_levels[] =
     { "5.1", "5.1", MFX_LEVEL_AV1_51,  },
     { "5.2", "5.2", MFX_LEVEL_AV1_52,  },
     { "5.3", "5.3", MFX_LEVEL_AV1_53,  },
-    { "6.0", "6.0", MFX_LEVEL_AV1_6,   },
-    { "6.1", "6.1", MFX_LEVEL_AV1_61,  },
-    { "6.2", "6.2", MFX_LEVEL_AV1_62,  },
-    { "6.3", "6.3", MFX_LEVEL_AV1_63,  },
     { NULL,                            },
+};
+static const char * const hb_av1_qsv_level_names[] =
+{
+    "auto", "2.0", "2.1", "2.2", "2.3", "3.0", "3.1", "3.2",
+    "3.3", "4.0", "4.1", "4.2", "4.3", "5.0", "5.1", "5.2",
+    "5.3", NULL,
 };
 
 static const enum AVPixelFormat hb_qsv_pix_fmts[] =
@@ -3170,10 +3193,10 @@ const char* const* hb_qsv_profile_get_names(int encoder)
         case HB_VCODEC_QSV_H265_8BIT:
             return hb_h265_profile_names_8bit;
         case HB_VCODEC_QSV_H265_10BIT:
-            return hb_h265_qsv_profile_names_10bit;
+            return hb_qsv_h265_profiles_names_10bit;
         case HB_VCODEC_QSV_AV1_10BIT:
         case HB_VCODEC_QSV_AV1:
-            return hb_av1_qsv_profile_names;
+            return hb_qsv_av1_profiles_names;
         default:
             return NULL;
     }
@@ -3184,10 +3207,10 @@ const char* const* hb_qsv_level_get_names(int encoder)
     switch (encoder)
     {
         case HB_VCODEC_QSV_H264:
-            return hb_h264_level_names;
+            return hb_h264_qsv_level_names;
         case HB_VCODEC_QSV_H265_10BIT:
         case HB_VCODEC_QSV_H265:
-            return hb_h265_level_names;
+            return hb_h265_qsv_level_names;
         case HB_VCODEC_QSV_AV1_10BIT:
         case HB_VCODEC_QSV_AV1:
             return hb_av1_qsv_level_names;

